docs/pages/guides/schema-validation.mdx

```diff
@@ -0,0 +1,120 @@
+
+The issue is that storage with Liveblocks contains a different type structure than your application (client)- which could break your application. How do you avoid this situation? By enforcing schema validation. While optional, schema validation is a vital tool for applications in a production environment- it ensures that errors like the one described above can be avoided and resolved while working in development mode. We’ll elaborate a bit more on the inner workings of schema validation below.
+
+## How schema validation works
+As described above, the primary purpose of schema validation is to prevent corrupted data from being loaded from the server. Two things need to happen to keep our data in sync and eliminate the risk of errors. First, we must ensure that data loaded into a room matches the underlying storage Tree and defined schema (if it exists). Second, we will introduce the rule that each mutation that occurs to their storage data will either be accepted or rejected.  
+
+When a room loads, it determines whether a schema exists; if it does, the value of the schema is parsed and assigned to the room.
+
+<!-- Insert an animation here !-->
+
+If the room does not enforce schema validation, all mutations in the room are accepted. 
+For example, imagine that a client performs an operation that updates the storage tree: regardless of whether or not this could break the application, Liveblocks has no opinion on what this mutation represents and accepts the incoming operation and synchronizes it with other clients. There is no right or wrong way to mutate data. After the operation is completed, the updated storage tree is forwarded to other users in the room. 
+
+When a room enforces schema validation and the client updates values in the storage, the operation is sent to the Liveblocks server, verifying that the updated Storage Tree is consistent with the defined schema attached to the room. If the verification succeeds, the storage data gets updated and synchronized with other users (clients) in the room. If the schema validation fails, the mutation gets rejected: storage will not be updated, other clients will never see the mutation, and the client that made the mutation will throw an error, and the developer can fix it. Situations like this should only happen in development mode, and the issue should be fixed by the developer, similar to fixing a TypeScript error before publishing it to production.
+
+## Create a schema
+To create a schema, you will need to POST to the schemas endpoint:
+```ts
+POST https://api.liveblocks.io/v2/schemas
+{
+  "id": "some-id",
+  "name": "Your schema name",
+  "data": `{
+    type Storage {
+      // your data schema will go here
+    }
+  }`,
+}
+
+```
+Note that the data type will be of type string because the schema will be parsed. For more information, see our [Schema Language documentation](xxx).
```

Comment:
After re reading the flow of the guide, I think having a schema language specification might be great for the launch. I can help you with this.

Reply:
Ok.  I know we have the details in the design spec- is there much deviation from this?

Reply:
Agreed, having that language specification would be fantastic! Beware of using the spec for this, though @taylorpwhipp! The language set we offer in the private beta is a lot more limited than the "vision" formulated in the spec yet. We'll be working on growing the supported language constructs, but most of that will happen after the launch and while we're actively in private beta.